### PR TITLE
Fix transfer modal queries

### PR DESCRIPTION
**Pull request by holygits**
selects default asset on load and changes balance display with selected asset.

gif is super slow 🤷‍♀️
Screen Recording 2020-12-01 at 17 26 51

### CODE REVIEW

File: packages/react-components/src/InputBalance.tsx [hunk 1]
```diff
@@ -35,7 +35,8 @@ interface Props extends BareProps {
 const DEFAULT_BITLENGTH = BitLengthOption.CHAIN_SPEC as BitLength;
 
 function InputBalance ({ autoFocus, className, defaultValue, help, isDisabled, isError, isFull, isZeroable, label, labelExtra, maxValue, onChange, onEnter, onEscape, placeholder, style, value, withEllipsis, withLabel, withMax }: Props): React.ReactElement<Props> {
-  const formattedDefaultValue =  defaultValue ? toFormattedBalance({ value: defaultValue }) : '0';
+  const formattedDefaultValue =  defaultValue ? toFormattedBalance({ value: defaultValue }) : "";
```
Comment (holygits) [inline packages/react-components/src/InputBalance.tsx:38]: add some better defaults
amount can be nothing "" and placeholder default is `0.0`

File: packages/app-accounts/src/modals/TransferWithType.tsx [hunk 2]
```diff
@@ -6,62 +6,68 @@ import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
...
-      value: assetId.toString()
-    };
-  });
+  const [assetBalance, setAssetBalance] = useState<BN>(new BN(0));
 
+  // Query registered assets on first load
   useEffect((): void => {
-    setOptions(dropdownOptions || []);
-  }, [retrivedAssetList]);
+    // @ts-ignore
```
Comment (alexsednz) [inline packages/app-accounts/src/modals/TransferWithType.tsx:47]: Is that necessary?

File: packages/app-accounts/src/modals/TransferWithType.tsx [hunk 3]
```diff
@@ -6,62 +6,68 @@ import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
...
+    api.rpc.genericAsset.registeredAssets().then(
+      (assets: any) => {
```
Comment (alexsednz) [inline packages/app-accounts/src/modals/TransferWithType.tsx:49]: Can we avoid "any" here?

File: packages/app-accounts/src/modals/TransferWithType.tsx [hunk 4]
```diff
@@ -6,62 +6,68 @@ import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
...
+        setAssetId(dropdownOptions[0].value || "0");
+        setAssetName(dropdownOptions[0].text || "?");
+      });
+  }, []);
+
+  // Query balances on assetId or senderId change
+  useMemo((): void => {
+    // @ts-ignore
+    api.query.genericAsset.freeBalance(assetId, senderId || "").then(
+      (balance: any) => setAssetBalance((balance as Balance).toBn())
```
Comment (alexsednz) [inline packages/app-accounts/src/modals/TransferWithType.tsx:68]: Same for "any" here

File: packages/app-accounts/src/modals/TransferWithType.tsx [hunk 5]
```diff
@@ -6,62 +6,68 @@ import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
...
+
+        setOptions(dropdownOptions);
+        setAssetId(dropdownOptions[0].value || "0");
+        setAssetName(dropdownOptions[0].text || "?");
+      });
+  }, []);
+
+  // Query balances on assetId or senderId change
+  useMemo((): void => {
+    // @ts-ignore
```
Comment (alexsednz) [inline packages/app-accounts/src/modals/TransferWithType.tsx:66]: Same for "ignore" here

File: packages/app-accounts/src/modals/TransferWithType.tsx [hunk 6]
```diff
@@ -77,6 +83,12 @@ function TransferWithType ({ assets, className, onClose, recipientId: propRecipi
     );
   }, [amount, assetId, recipientId, senderId]);
 
+  // When assetId is selected, update assetName also
+  function setAssetId_(assetId: string): void {
```
Comment (alexsednz) [inline packages/app-accounts/src/modals/TransferWithType.tsx:87]: Perhaps a name like setAssetIdName or setAssetInfo would be a better option.

File: packages/apps/public/locales/en/app-accounts.json [hunk 7]
```diff
@@ -143,5 +143,28 @@
...
+  "You will need this": "You will need this",
+  "password + your keystore file (download next step)": "password + your keystore file (download next step)",
+  " to unlock your CENNZnet account.": " to unlock your CENNZnet account.",
+  "Back it up on paper or hard drive.": "Back it up on paper or hard drive.",
+  "send from account": "send from account",
+  "send to address": "send to address",
+  "Select the asset you want to transfer.": "Select the asset you want to transfer.",
+  "Asset type": "Asset type",
+  "Enter the amount you want to transfer.": "Enter the amount you want to transfer.",
+  "Send amount": "Send amount"
```
Comment (alexsednz) [inline packages/apps/public/locales/en/app-accounts.json:169]: Could we choose shorter names for some of the above messages?

File: packages/apps/public/locales/en/ui.json [hunk 8]
```diff
@@ -872,5 +872,31 @@
   "Uncaught error. Something went wrong with the query and rendering of this component.": "",
   "validators/nominators": "",
   "(Advanced) Select the signature scheme for this account.": "",
-  "Advanced": ""
```
Comment (KarishmaBothara) [inline packages/apps/public/locales/en/ui.json:875]: I think may be we don't want to include all the json files in locales/en/*

File: packages/react-components/src/InputNumber.tsx [hunk 9]
```diff
@@ -197,13 +197,6 @@ export default function InputNumber (props: Props): React.ReactElement<Props> {
     }
   };
 
-  const _onPaste = (event: React.ClipboardEvent<Element>): void => {
```
Comment (holygits) [inline packages/react-components/src/InputNumber.tsx:200]: regex here was preventing paste

File: packages/app-accounts/src/modals/TransferWithType.tsx (hunk 4 above)
Comment (KarishmaBothara) [reply packages/app-accounts/src/modals/TransferWithType.tsx:81]: may be we can use (balance: Balance) here

**Review by KarishmaBothara — APPROVED** [review]

**Pull request merged**